hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/AbstractInternalQueryCache.java

```diff
@@ -72,8 +73,9 @@ public AbstractInternalQueryCache(String cacheId, String cacheName, IMap delegat
         this.serializationService = context.getSerializationService();
         // We are not using injected index provider since we're not supporting off-heap indexes in CQC due
         // to threading incompatibility. If we injected the IndexProvider from the MapServiceContext
-        // the EE side would create HD indexes which is undesired
-        this.indexes = new Indexes(serializationService, new DefaultIndexProvider(), Extractors.empty(), true);
+        // the EE side would create HD indexes which is undesired.
+        this.indexes = new Indexes(serializationService, new DefaultIndexProvider(), Extractors.empty(), true,
+                IndexCopyBehavior.NEVER);
```

Comment:
How come this is `IndexCopyBehavior.NEVER` here? Wasn't the old behaviour copy-on-read? It's not needed anymore?

Reply:
Well, I just made an assumption that the QueryCache is not accurate anyway, since it's local to the node, so I just accepted this trade-off. But you are right that it may change the behavior, so probably I should not make this call here.

Reply:
I didn't go into the behaviour of the query cache. Perhaps `NEVER` is an acceptable option. Maybe someone with query cache knowledge can decide if this is a good and acceptable performance boost - @ahmetmircik

Reply:
`COPY_ON_READ` seems safer.

Reply:
FIxed to COPY_ON_READ a while ago. Thx